stdlib/public/Concurrency/Task.cpp

```diff
@@ -642,6 +642,8 @@ static AsyncTaskAndContext swift_task_create_commonImpl(
     swift_retain(task);
     swift_task_enqueue(task, executor);
   }
+  
+  _swift_tsan_release(static_cast<Job *>(task));
```

Comment:
Actually, the false positive went away again after https://github.com/apple/swift/commit/3727db2fe2f3bd9c1f4b75ba3343c5445de53740 because the call to `swift_task_enqueue()` adds the required release edge.

@DougGregor 
Are all tasks that are created here, i.e., also the ones that are not immediately enqueued above (`taskCreateFlags.enqueueJob() == false`) always guaranteed to be scheduled via `swift_task_enqueue()` before they start executing on a different thread?  If so, then we don't need this anymore.

Reply:
They will be scheduled via `swift_task_enqueue` at some point, yes.

Reply:
Do you know what changed here?  Previously, not all tasks were going through the `swift_task_enqueue()` funnel, which produced the false positive we were seeing.

Do you recommend adding this (superfluous) edge here to protect us from future changes, or relying on "all tasks go through `swift_task_enqueue ()` even if not immediately scheduled"?

Reply:
Doug pointed out that some places call `swift_task_enqueueGlobal()` directly.

The most common (and currently covered) case seems to be `swift_task_enqueue()` calling `swift_task_enqueueGlobal()`, but if the latter (more specialized) version is getting called “by itself”, we should add an edge there (and actually not at task creation).

Reply:
I moved the TSan release edge from `swift_task_create_commonImpl()` to `swift_task_enqueueGlobalImpl()`.  Task creation itself is not an event that needs synchronization, but rather that task creation "happens before" execution of that task on another thread.

This edge is usually added when the task is scheduled via `swift_task_enqueue()` (which then usually calls `swift_task_enqueueGlobal()`).  However, not all task scheduling goes through the `swift_task_enqueue()` funnel: some places call the more specific `swift_task_enqueueGlobal()` directly.  So let's annotate this function (duplicate edges aren't harmful) to ensure we cover all schedule events, including newly-created tasks (our original problem).